04-onion-routing.md

```diff
@@ -263,6 +263,9 @@ It is formatted according to the Type-Length-Value format defined in [BOLT #1](0
     2. data:
         * [`32*byte`:`payment_secret`]
         * [`tu64`:`total_msat`]
+    1. type: 10 (`payment_metadata`)
```

Comment:
Updated

Reply:
Actually, it looks like AMP is using 14 already. Shall we move to 16 then?

Reply:
What about AMP updating to another field? It's not documented that they use 14, not even in #658 AFAICT...

Reply:
Haha, that is not a q for me to answer :popcorn: 

To be consequent, I think we should use 10 then?

Reply:
> To be consequent, I think we should use 10 then?

Not quite, route blinding has documented that it uses 10 and 12 (for 6+ months) and already has two implementations' ACKs and cross-compatibility successfully tested (even before `payment_metadata`, so it does take precedence). AMP has no spec and only one implementation...

Your call, you can either update to 16 here or ask lnd folks. In all cases using 14 for AMP without specifying it anywhere is reckless, it just creates a big risk that another feature will ship across multiple implementations and use 14 for something else which will be a huge pain to update. Until specified, AMP should use a TLV in the high range to avoid running into this kind of compatibility problems.

Reply:
Let's update to 16 so that we are unblocked.

Tagging @Roasbeef here for comment on AMP.